musica/mechanism_configuration/arrhenius.py

```diff
@@ -72,44 +72,216 @@ def __init__(
+        cpp_reactants = []
+        for r in value:
+            if isinstance(r, Species):
+                cpp_reactants.append(_ReactionComponent(r.name))
+            elif isinstance(r, tuple) and len(r) == 2:
+                coefficient, species = r
+                cpp_reactants.append(_ReactionComponent(species.name, coefficient))
+            else:
+                raise ValueError(f"Invalid reactant format: {r}")
+        self._instance.reactants = cpp_reactants
+    
+    @property
+    def products(self) -> List[Union[Species, Tuple[float, Species]]]:
+        """Get the products as Python objects."""
+        # Convert from C++ _ReactionComponent objects to Python Species objects
+        result = []
+        for rc in self._instance.products:
+            if hasattr(rc, 'coefficient') and rc.coefficient != 1.0:
+                # Create a tuple with coefficient and species
+                species = Species(name=rc.species_name)
+                result.append((rc.coefficient, species))
+            else:
+                # Just the species
+                species = Species(name=rc.species_name)
+                result.append(species)
+        return result
+    
+    @products.setter
+    def products(self, value: List[Union[Species, Tuple[float, Species]]]):
+        """Set the products, converting from Python to C++ objects."""
+        cpp_products = []
+        for p in value:
+            if isinstance(p, Species):
+                cpp_products.append(_ReactionComponent(p.name))
+            elif isinstance(p, tuple) and len(p) == 2:
+                coefficient, species = p
+                cpp_products.append(_ReactionComponent(species.name, coefficient))
+            else:
+                raise ValueError(f"Invalid product format: {p}")
+        self._instance.products = cpp_products
+    
+    @property
+    def gas_phase(self) -> str:
+        """Get the gas phase name."""
+        return self._instance.gas_phase
+    
+    @gas_phase.setter
+    def gas_phase(self, value: Union[Phase, str]):
+        """Set the gas phase."""
+        if isinstance(value, Phase):
+            self._instance.gas_phase = value.name
+        elif isinstance(value, str):
+            self._instance.gas_phase = value
+        else:
+            raise ValueError(f"Invalid gas_phase type: {type(value)}")
+    
+    @property
+    def other_properties(self) -> Dict[str, Any]:
+        """Get the other properties."""
+        return self._instance.other_properties
+    
+    @other_properties.setter
+    def other_properties(self, value: Dict[str, Any]):
+        """Set the other properties."""
+        self._instance.other_properties = value
+
+    def serialize(self) -> Dict:
+        """
+        Serialize the Arrhenius object to a dictionary using only Python-visible data.
+        
+        Returns:
+            Dict: A dictionary representation of the Arrhenius object.
+        """
+        serialize_dict = {
```

Comment:
This dictionary is the same as in the `serialize_static` function. Put that in one place so it isn't duplicated